src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,868 @@
+    uint256 public constant MAX_PENDING_UPDATES = 1;
+
+    /// @dev Returns the chain ID from the time the contract was deployed.
+    uint256 internal immutable ORIGINAL_CHAIN_ID;
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+
+    /**
+     * @dev Initializes the immutable addresses of the strategy mananger, delegationManager, slasher,
+     * and eigenpodManager contracts
+     */
+    constructor(
+        IDelegationManager _delegation,
+        IAVSDirectory _avsDirectory,
+        uint32 _DEALLOCATION_DELAY,
+        uint32 _ALLOCATION_DELAY_CONFIGURATION_DELAY
+    )
+        AllocationManagerStorage(_delegation, _avsDirectory, _DEALLOCATION_DELAY, _ALLOCATION_DELAY_CONFIGURATION_DELAY)
+    {
+        _disableInitializers();
+        ORIGINAL_CHAIN_ID = block.chainid;
+        DEALLOCATION_DELAY = _DEALLOCATION_DELAY;
+    }
+
+    /**
+     * @dev Initializes the addresses of the initial owner, pauser registry, and paused status.
+     * minWithdrawalDelayBlocks is set only once here
+     */
+    function initialize(
+        address initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 initialPausedStatus
+    ) external initializer {
+        _initializePauser(_pauserRegistry, initialPausedStatus);
+        _DOMAIN_SEPARATOR = _calculateDomainSeparator();
+        _transferOwnership(initialOwner);
+    }
+
+    /**
+     * @notice Called by the delagation manager to set delay when operators register.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     * @dev msg.sender is assumed to be the delegation manager.
+     */
+    function setAllocationDelay(address operator, uint32 delay) external {
+        require(msg.sender == address(delegation), OnlyDelegationManager());
+        _setAllocationDelay(operator, delay);
+    }
+
+    /**
+     * @notice Called by operators to set their allocation delay.
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     */
+    function setAllocationDelay(
+        uint32 delay
+    ) external {
+        require(delegation.isOperator(msg.sender), OperatorNotRegistered());
+        _setAllocationDelay(msg.sender, delay);
+    }
+
+    /**
+     * @notice For all pending deallocations that have become completable, their pending free magnitude can be
+     * added back to the free magnitude of the (operator, strategy) amount. This function takes a list of strategies
+     * and adds all completable deallocations for each strategy, updating the freeMagnitudes of the operator
+     *
+     * @param operator address to complete deallocations for
+     * @param strategies a list of strategies to complete deallocations for
+     * @param numToComplete a list of number of pending free magnitude deallocations to complete for each strategy
```

Comment:
two nits:

1. `adds all` is somewhat technically inaccurate, since there is a `numToComplete` input
2. this should probably specify that it's fine (i.e. function won't revert) if `numToComplete` exceeds the actual completable number -- e.g. just using type(uint16).max as the input shouldn't revert except for running out of gas

Reply:
Love the natspec corrections, rn natspec isn't the best. 

Will reach out to relevant parties to get these improved.